hack/update-codegen.sh

```diff
@@ -1,36 +1,29 @@
-#!/bin/bash -xe
+#!/bin/bash
 
-SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../../../k8s.io/code-generator)}
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
 
-# use gsed for MAC env
-SED_CMD=sed
-if [[ `uname` == 'Darwin' ]]; then
-  SED_CMD=gsed
-fi
+REPO_ROOT=$(realpath $(dirname ${BASH_SOURCE[0]})/..)
 
-# HACK: For some reason this script is not executable.
-${SED_CMD} -i 's,^exec \(.*/generate-internal-groups.sh\),bash \1,g' ${CODEGEN_PKG}/generate-groups.sh
-# ...but we have to put it back, or `verify` will puke.
-trap "git checkout ${CODEGEN_PKG}/generate-groups.sh" EXIT
+GO111MODULE=on go install k8s.io/code-generator/cmd/deepcopy-gen@release-1.29
```

Comment:
I ran into the same failure with `go install <web URI>` in https://github.com/openshift/hive/pull/2481

Can we try:
- make sure we have code-generator vendored at the appropriate version in go.mod, and naked-imported in pkg/dependencymagnet/go.doc
- Use `go install $(shell realpath vendor/k8s.io/code-generator/cmd/deepcopy-gen` here to build the tool from the vendor dir

?